lib/FreeRTOS/tasks.c

```diff
@@ -3980,7 +3980,7 @@ TCB_t *pxTCB;
 				{
 					if( uxListRemove( &( pxMutexHolderTCB->xStateListItem ) ) == ( UBaseType_t ) 0 )
 					{
-						taskRESET_READY_PRIORITY( pxMutexHolderTCB->uxPriority );
+						portRESET_READY_PRIORITY( pxMutexHolderTCB->uxPriority );
```

Comment:
As mentioned on one of your other PRs, a task may have been removed from delayed list and therefore a check of ready list is needed before calling ```portRESET_READY_PRIORITY```. This change, therefore, cannot be done. 

Thanks.

Reply:
But in this case, before we call taskRESET_READY_PRIORITY, we can confirm it must be in the ready list, 
![image](https://user-images.githubusercontent.com/29007810/59153568-9f0e1480-8a8e-11e9-956e-a8d87eb82698.png)

Reply:
Yes, I missed that and you are right. Thanks for pointing it out. We will make this update in the master copy of the kernel, so it won't be visible here for a while. 

Thanks.

Reply:
You're welcome.
By the way, the same issue happen in another place of task.c:
task.c  -- vTaskPriorityDisinheritAfterTimeout  --  Line 4185